lib/ruby_indexer/lib/ruby_indexer/index.rb

```diff
@@ -360,6 +361,14 @@ def resolve(name, nesting, seen_names = [])
       ).void
     end
     def index_all(indexable_paths: @configuration.indexables, &block)
+      # When troubleshooting an indexing issue, e.g. through irb, it's not obvious that `index_all` will augment the
+      # existing index values, meaning it may contain 'stale' entries. This check ensures that the user is aware of this
+      # behavior and can take appropriate action.
+      # binding.break
+      if @entries.any?
+        raise IndexNotEmptyError, "The index is not empty. Create a new instance before re-indexing"
+      end
```

Comment:
Isn't this going to fail every single time when using the LSP itself because we always index RBS first?